stock_reserve/model/stock_reserve.py

```diff
@@ -56,28 +56,65 @@ class StockReservation(models.Model):
         required=True,
         readonly=True,
         ondelete='cascade',
-        select=1)
+        index=True)
     date_validity = fields.Date('Validity Date')
 
     @api.model
     def default_get(self, fields_list):
+        """ Fix default values
+
+            - Ensure default value of computed field `product_qty` is not set
+              as it would raise an error
+            - Compute default `location_id` based on default `picking_type_id`.
+              Note: `default_picking_type_id` may be present in context,
+              so code that looks for default `location_id` is implemented here,
+              because it relies on already calculated default
+              `picking_type_id`.
         """
-        Ensure default value of computed field `product_qty` is not set
-        as it would raise an error
-        """
+        # if there is 'location_id' field requested, ensure that
+        # picking_type_id is also requested, because it is required
+        # to compute location_id
+        if ('location_id' in fields_list and
+                'picking_type_id' not in fields_list):
+            fields_list = fields_list + ['picking_type_id']
+
         res = super(StockReservation, self).default_get(fields_list)
+
         if 'product_qty' in res:
             del res['product_qty']
+
+        # At this point picking_type_id and location_id
+        # should be computed in default way:
+        #     1. look up context
+        #     2. look up ir_values
+        #     3. look up property fields
+        #     4. look up field.default
+        #     5. delegate to parent model
+        #
+        # If picking_type_id is present and location_id is not, try to find
+        # default value for location_id
+        if not res.get('picking_type_id', None):
+            res['picking_type_id'] = self._default_picking_type_id()
+
+        picking_type_id = res.get('picking_type_id', None)
```

Comment:
`.get` implicitly returns `None` when a key is not existing. `False` would only be returned in the event that the key exists, and is simply not set in the data. This would be the same as if directly accessing via `res.picking_type_id`

That said, the `None` here is superfluous & can be removed.